joschahenningsen/TUM-Live
# course management data-providers refactor

### Motivation and Context
Currently, the course management relies on Go generated template data. This means we need page reloads
and also it is not good practice to have backend and frontend entangled in that way. Further, much very specific code is used to handle the course management tasks, that could be abstracted in cleaner ways. 

### Description
This PR untangles backend and frontend by using data-providers that are introduced earlier and are already used for data-sections and other parts of the app. This gives a standardized abstraction to handle the lecture's client side.
Also, this will prevent useless page reloads and give a better standardized and app-global API to create new lectures, fetch or update them.

### Steps for Testing
- Go to course management Page
- Open a course
- Edit a lecture
- a) Check uploading media
- b) Check changing meta data
- c) Check uploading attachments (Drag and drop Description field)
- Check JS-Console and Network Tab for errors
- Reload and check if data is persistent

### Screenshots

Way shorter UI specific code (edit-course.ts):

![difference](https://github.com/TUM-Dev/gocast/assets/17506411/81c2978c-d09b-48cc-917c-c611264efd57)
